lib/omniauth/strategies/openid_connect.rb

```diff
@@ -107,7 +108,7 @@ def request_phase
       def callback_phase
         error = params['error_reason'] || params['error']
         error_description = params['error_description'] || params['error_reason']
-        invalid_state = params['state'].to_s.empty? || params['state'] != stored_state
+        invalid_state = options.verify_state && (params['state'].to_s.empty? || params['state'] != stored_state)
```

Comment:
I'd not exclude state mismatch from the check. Only a case when there is no state stored.

Reply:
Fair - made the changes.